### PR TITLE
Mapping API - Refactoring

### PR DESCRIPTION
**Pull request by eirsyl**

### CODE REVIEW

**Comment by eirsyl** [comment]
API for restricted and normal mappings has a attribute 
```
"recipient_list": [
            2
        ]
```
This contains only pks for the recipients, to get the recipients the agent needs to do n gets, maybe we should fill this list with objects containing the recipient address, tag and pk?

File: holonet/restricted/models.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class RestrictedMapping(TokenModel):
 
     recipient_list = models.ManyToManyField('mappings.Recipient', blank=True,
                                             related_name='restricted_lists')
-    tag = models.CharField(max_length=100, blank=True)
+    tag = models.CharField(max_length=100, unique=True, null=True)
```
Comment (relekang) [inline holonet/restricted/models.py]: CharFields shouldn't have null=True (https://docs.djangoproject.com/en/1.7/ref/models/fields/#null)

File: holonet/mappings/models.py [hunk 2]
```diff
@@ -22,20 +22,23 @@ def recipients(self):
...
-                raise ValueError('Tag field is not unique.')
+        if self.tag == '':
+            raise IntegrityError('The tag field cannot be a empty string.')
         return super(MailingList, self).save(*args, **kwargs)
 
 
 class Recipient(models.Model):
     address = models.EmailField()
-    tag = models.CharField(max_length=100, unique=True)
+    tag = models.CharField(max_length=100, unique=True, null=True)
```
Comment (relekang) [inline holonet/mappings/models.py]: CharFields shouldn't have null=True (https://docs.djangoproject.com/en/1.7/ref/models/fields/#null)

File: holonet/mappings/models.py [hunk 3]
```diff
@@ -12,7 +12,7 @@ class MailingList(models.Model):
 
     recipient_list = models.ManyToManyField('mappings.Recipient', blank=True,
                                             related_name='mailing_lists')
-    tag = models.CharField(max_length=100, blank=True)
+    tag = models.CharField(max_length=100, unique=True, null=True)
```
Comment (relekang) [inline holonet/mappings/models.py]: CharFields shouldn't have null=True (https://docs.djangoproject.com/en/1.7/ref/models/fields/#null)

**Comment by relekang** [comment]
> This contains only pks for the recipients, to get the recipients the agent needs to do n gets, maybe we should fill this list with objects containing the recipient address, tag and pk?

Yes, that is a good idea :)

**Comment by eirsyl** [comment]
We need tests for recipient and mapping apis

**Pull request merged**